### PR TITLE
kubernetes-code-generator: init at 0.25.4

### PR DESCRIPTION
**Pull request by urandom2**
###### Project description
Generators for kube-like API types

###### Metadata

- homepage URL: https://github.com/kubernetes/code-generator
- source URL: https://github.com/kubernetes/code-generator
- license: Apache License 2.0
- platforms: any

##### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/code-generator/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildGoModule, fetchFromGitHub, git }:
```
Comment (muscaln) [inline pkgs/applications/networking/cluster/code-generator/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/applications/networking/cluster/code-generator/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "kubernetes";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-GKF6DXvyZujInOZbV0ePUu71BEl1s/chNTN1PucdIYw=";
+  };
+
+  vendorHash = "sha256-zjgTtGen6a8TPi/DrwheTS1VQ+hd+KI7UHoyMZ4W4+k=";
+
```
Comment (muscaln) [inline pkgs/applications/networking/cluster/code-generator/default.nix:15]:
```suggestion

  ldflags = [ "-s" "-w" ];

```

**Review by muscaln — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/code-generator/default.nix (hunk 2 above)
Comment (urandom2) [reply pkgs/applications/networking/cluster/code-generator/default.nix:15]: I have only added `ldflags` when I want to do a `-X path/to/pkg.symbol=value`, is this required by default now? (and if so, why is it not the default when you leave `ldflags = null`)

Comment (azahi) [reply pkgs/applications/networking/cluster/code-generator/default.nix:15]: See #177698.

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 203926` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubernetes-code-generator</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

**Pull request merged**